### ISSUE TITLE
Some weapons Softlocks

### ISSUE THREAD

**Issue by Niamek** [abyssonym/smrpg_gbarp#2]
When using Superslap with Geno, the game softlocked. Couldn't do anything.

PS: I do not mind the mixed sprites during the animations, no need to fix that!

SEED:  1473554902

**Comment by Niamek** [comment]
![super mario rpg - legend of the seven stars usa 1473554902000](https://cloud.githubusercontent.com/assets/15385566/18414875/b5a3bbec-77a8-11e6-9dea-35ef06d9ddad.png)

Forgot to attach a screen. It stayed like that for a long time. Croc is still moving.

**Comment by abyssonym** [comment]
Thanks for letting me know. Character/weapon swapping seemed stable, so I didn't check all of the weapons. Lemme know if you find any other weapons that are broken.

**Comment by TheBuffalax** [comment]
https://www.twitch.tv/buffalax/v/89976576 i had the same thing happen. it occurred near the end of the clip

the weapon was the handgun

**Comment by Niamek** [comment]
Geno with Super hammer softlocks too.

![smrpg 1474329533000](https://cloud.githubusercontent.com/assets/15385566/18654694/2c46ba74-7eb1-11e6-836d-90e0beb25a34.png)

**Comment by abyssonym** [comment]
Unfortunately, this issue seems to be specific to character-weapon pairs. Geno + Super Hammer freezes, but Mallow + Super Hammer works. I haven't tested every combination, but I tested everything on Mario and Mallow, and banned the following weapons from being equipped on different characters:

```
Super Hammer
Handgun
Double Punch
Super Slap
```

I'm leaving this issue open so that other people can report other weapons if necessary.

**Comment by Niamek** [comment]
Oh that's too bad. Maybe just restrict who can equipe X weapon? For example for the Super Hammer, never allow Geno to equipe it? Like a case by case approach?

I believe you should do that because if you ban some weapons from being equipped to different characters, randomization is reduced by a lot.

**Comment by MetroidMst** [comment]
Just wanted to mention I have come across this issue with a combination I didn't see mentioned.

My combination is Geno - NokNok Shell
Also got another Geno combinations with the Frying Pan that doesn't work too.

**Comment by abyssonym** [comment]
Thanks. The common theme seems to be Geno, so I'll try looking at him next.

**Comment by Niamek** [comment]
If it's not fixable, maybe just remove Geno and his weapon from randomization?

**Comment by abyssonym** [comment]
As of version 3, I've decided not to randomize who can equip Geno's weapons or give him new weapons to equip. This was the safest and easiest solution and seemed the most appropriate since there were so many issues with him and his weapons; I don't want to find out down the line that a Geno combination that looked safe actually isn't in some circumstance. Please let me know if there are any other combinations not involving Geno that cause crashes. I'll leave this issue open a while longer.

**Comment by Niamek** [comment]
Did you randomize the other weapons mentionned like Super hammer? I think it will be fine if you randomize them, but away from Geno.

**Comment by abyssonym** [comment]
Yeah, all other weapons are randomized, just not to or from Geno.

**Comment by Niamek** [comment]
Awesome. :+1:

**Comment by Niamek** [comment]
I would like to know. Are the stats of the weapons still randomized? I'm currently playing a seed and the weapons stats aren't as random as it was before (In past versions, I saw some weapons messing the special attacks or even defense).
